x/application/keeper/auto_undelegate.go

```diff
@@ -0,0 +1,73 @@
+package keeper
+
+import (
+	"fmt"
+	"slices"
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const eventGatewayUnstaked = "poktroll.gateway.GatewayUnstaked"
```

Comment:
Can this be retrieved programatically with something like:
```suggestion
var eventGatewayUnstaked = new(applicationtypes.EventGatewayUnstaked).Name()"
```


Otherwise:
```suggestion
const eventGatewayUnstaked = "poktroll.gateway.EventGatewayUnstaked"
```